it/controllers/StartDateControllerISpec.scala

```diff
@@ -472,7 +472,141 @@ class StartDateControllerISpec extends ControllerITTestHelper {
         }
       }
     }
+
     testUnauthorisedUser(baseUrl + normalRoutePath, Some(Json.obj("value" -> "true")))
     testAuthenticatedUserButNoUserAnswers(baseUrl + normalRoutePath, Some(Json.obj("value" -> "true")))
   }
+  "Post in either normal mode or check mode " - {
+    "Should redirect to the Pack-At-Business address controller when the user is a large producer and has answered Yes " +
+      "to either Operate Packaging Sites or Contract Packer" in {
+      given
+        .commonPrecondition
+
+      setAnswers(largeProducerUserAnswers)
+      WsTestClient.withClient { client =>
+
+        val result = createClientRequestPOST(
+          client, baseUrl + checkRoutePath, validDateJson
+        )
+
+        whenReady(result) { res =>
+          res.status mustBe 303
+          res.header(HeaderNames.LOCATION) mustBe Some(routes.PackAtBusinessAddressController.onPageLoad(CheckMode).url)
+          val dataStoredForPage = getAnswers(identifier).fold[Option[LocalDate]](None)(_.get(StartDatePage))
+          dataStoredForPage.nonEmpty mustBe true
+          dataStoredForPage.get mustBe date
+        }
+      }
+    }
+
+    "Should redirect to the Pack-At-Business address controller when the user is a small producer and has answered Yes " +
+      "to Contract Packer" in {
+      given
+        .commonPrecondition
+
+      setAnswers(smallProducerUserAnswers)
+      WsTestClient.withClient { client =>
+
+        val result = createClientRequestPOST(
+          client, baseUrl + checkRoutePath, validDateJson
+        )
+
+        whenReady(result) { res =>
+          res.status mustBe 303
+          res.header(HeaderNames.LOCATION) mustBe Some(routes.PackAtBusinessAddressController.onPageLoad(CheckMode).url)
+          val dataStoredForPage = getAnswers(identifier).fold[Option[LocalDate]](None)(_.get(StartDatePage))
+          dataStoredForPage.nonEmpty mustBe true
+          dataStoredForPage.get mustBe date
+        }
+      }
+    }
+
+    "Should redirect to the Pack-At-Business address controller when the user is a non producer and has answered Yes " +
+      "to Contract Packer" in {
+      given
+        .commonPrecondition
+
+      setAnswers(smallProducerUserAnswers)
+      WsTestClient.withClient { client =>
+
+        val result = createClientRequestPOST(
+          client, baseUrl + checkRoutePath, validDateJson
+        )
+
+        whenReady(result) { res =>
+          res.status mustBe 303
+          res.header(HeaderNames.LOCATION) mustBe Some(routes.PackAtBusinessAddressController.onPageLoad(CheckMode).url)
+          val dataStoredForPage = getAnswers(identifier).fold[Option[LocalDate]](None)(_.get(StartDatePage))
+          dataStoredForPage.nonEmpty mustBe true
+          dataStoredForPage.get mustBe date
+        }
+      }
+    }
+
+    "Should redirect to the Pack-At-Business address controller when the user is a large producer and has answered No " +
```

Comment:
I think this test name is incorrect, is redirects to ask secondary warehouse,

Reply:
Good spot.  Corrected